service/src/main/kotlin/xyz/lebkuchenfm/domain/xsounds/XSoundsService.kt

```diff
@@ -24,10 +24,10 @@ class XSoundsService(private val repository: XSoundsRepository, private val file
         return repository.findByName(soundName)
     }
 
-    suspend fun incrementPlayCount(soundName: String) {
+    suspend fun markAsPlayed(soundName: String) {
         val xSound = this.getByName(soundName)
         xSound?.let {
-            repository.replace(it.copy(timesPlayed = it.timesPlayed + 1))
+            repository.incrementPlayCount(it)
```

Comment:
Check if the `incrementPlayCount` returns null. If it does we would like to log that we couldn't increment the play count for some reason.